apollo-compiler/src/test/graphql/com/example/simple_inline_fragment/TestQuery.java

```diff
@@ -335,10 +323,15 @@ public AsDroid read(String conditionalType, ResponseReader reader) throws IOExce
 
         @Override
         public Hero map(ResponseReader reader) throws IOException {
-          final String name = reader.read(fields[0]);
-          final AsHuman asHuman = reader.read(fields[1]);
-          final AsDroid asDroid = reader.read(fields[2]);
-          return new Hero(name, asHuman, asDroid);
+          final AsHuman asHuman = reader.read(fields[0]);
```

Comment:
What would happen if I specified two fragments that acted on Droid? This seems like it would return the first one, making the second one in accessible